core/src/main/java/com/linecorp/armeria/server/streaming/ServerSentEvents.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.streaming;
+
+import static com.linecorp.armeria.internal.ResponseConversionUtil.streamingFrom;
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.Executor;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.sse.ServerSentEvent;
+
+/**
+ * A utility class which helps to create a Server-sent Event stream from a content {@link Publisher}
+ * or {@link Stream}.
+ *
+ * @see <a href="https://www.w3.org/TR/eventsource/">Server-sent Event</a>
+ */
+public final class ServerSentEvents {
```

Comment:
How about accepting:

- `Publisher<? extends ServerSentEvent>` or `Stream<? extends ServerSentEvent>` instead of `Publisher/Stream<T>` when no converter is given, and
- `Publisher/Stream<T>` and `Function<T, ServerSentEvent>` when conversion is necessary,

.. instead of always converting an object into a data event?

If sending data-only stream is common, we could provide some out-of-the-box converter functions, e.g. `ServerSentEvents.newDataEvent(String)` -> `fromPublisher(stringPublisher, ServerSentEvents::newDataEvent)`

By doing this, we could reduce the ambiguity in the conversion logic at `toHttpData()`.

Reply:
Thanks! Fixed as you told. Please let me know if I missed your intention. :-)